concepts/80 Troubleshooting/05 DataGrid and TreeList Issues/10 Gray Boxes Appear After Data Loading/25 Eliminate Side Effects.md

```diff
@@ -0,0 +1,103 @@
+Follow the steps below to eliminate possible side effects:
+
+- Disable the [stateStoring](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/stateStoring/) property. Check if the issue still persists. If the issue disappears, it is possible that the saved state is corrupted. Clear the saved state, enable **stateStoring**, and check again.
+
+- FireFox and Safari browsers raise native scrolling events asynchronously. This behavior forces asynchronous row rendering in native scrolling mode even when [renderAsync](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/#renderAsync) is disabled. To avoid this side effect, disable the scrolling.[useNative](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/#useNative) property. Call the [defaultOptions(rule)](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#defaultOptionsrule) method to eliminate the issue for all DataGrid components in the application.
+
+- If you bind DataGrid or TreeList to one DataSource instance, reset the DataSource page index once you destroy the component. DataSource does not reset the index automatically. A new component bound to this DataSource may attempt to display the first data page while DataSource may contain only other data pages. The component displays gray boxes in such case.
+Call the DataSource.[pageIndex(newIndex)](/Documentation/ApiReference/Data_Layer/DataSource/Methods/#pageIndexnewIndex) method to reset the page index.
+
+---
+
+##### Angular
+
+- If you use conditional rendering to display DataGrid or TreeList, and virtual scrolling is enabled, the following happens:
+
+    - **on hide**    
+    The scroll position is discarded.
+
+    - **on show**    
+    The scroll position is set to the first page.
+
+    If DataGrid or TreeList was opened, for example, on the page 5 before it was hidden, the gray boxes are displayed after the component is shown for the second time. This happens because data source is loaded for the page 5 only.
```

Comment:
Well, data source is countable... I believe this case is definite because we imply the [current] data source:

```suggestion
    If DataGrid or TreeList was opened, for example, on the page 5 before it was hidden, the gray boxes are displayed after the component is shown for the second time. This happens because the data source is loaded for the page 5 only.
```